jhucourse/woj/backend/game.py

```diff
@@ -0,0 +1,134 @@
+'import json'
+import numpy
+import player
+'import QuestionBank'
+
+class game:
+    def __init__(self):
+        self.player_list = []
+        self.current_player = 0
+        self.spin_counter = 50
+        self.current_round = 1
+        self.turnPoints = 0
+        self.question_bank
+
+    'Add a player to the list'
+    def addPlayer(self, name):
+        Player = player(name)
+        self.player_list.append(Player)
+        return
+
+    'Remove a player from the list'
+    def deletePlayer(self, player):
+        self.player_list.remove(player)
+        return
+
+    'Get the next player in the list'
+    def nextPlayer(self):
+        len = self.player_list.count
+        current = self.current_player
+
+        if (current+1 >= len):
+            self.current_player = 0
+        else:
+            self.current_player = current+1
+
+        self.resetTurnPoints()
+        return
+
+    'Get the current player'
+    def getCurrentPlayer(self):
+        current = self.current_player
+        return self.player_list(current)
+
+    'Reset the points for current turn'
+    def resetTurnPoints(self):
+        self.turnPoints = 0
+        return
+
+    """
+    Update the points for the current turn
+    based on the point value and question result.
+    If the answer was correct (true) add points, else subtract
+    """
+    def updateTurnPoints(self, points, result):
+        turnPoints = self.calculateRoundPoints(points)
+
+        if (result):
+            self.turnPoints += turnPoints
+        else:
+            self.turnPoints -= turnPoints
+        return
+
+    'Calculate points based on the round'
+    def calculateRoundPoints(self, points):
+        return (points * self.current_round)
+
+    'Update the current player score'
+    def updatePlayerScore(self):
+        player = self.getCurrentPlayer()
+        points = self.calculatePoints()
+        player.set_player_score(points)
+        return
+
+    'Return the current number of wheel spins'
+    def getWheelSpins(self):
+        return self.spin_counter
+
+    """
+    Check the spins remaining on the wheel. 
+    If zero return true to indicate out of spins.
+    """
+    def verifyWheelSpins(self):
+        if (self.getWheelSpins <= 0):
+            return True
+        else:
+            return False
+    
+    'Reset the wheel spins back to 50'
+    def resetWheelSpins(self):
+        self.spin_counter = 50
+        return
+
+    'Decrease the wheel spins by 1'
+    def decreaseWheelSpins(self):
+        self.spin_counter -= 1
+        return
+
+    """
+    Check the number of unanswered questions.
+    If all questions have been answered return true.
+    """
+    def verifyQuestionCount(self):
```

Comment:
What is this doing?

Reply:
One of the ways to end the round is if we run out of questions. I don't know how we plan to store them, there've been talks about JSON and SQLite, but I figured we should incorporate a way to count the number of questions remaining since we'd need to keep track of what had been answered/viewed to retrieve the next question.